IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,468 @@
+
+In addition to the basic account & credential properties, the following definitions are used below:
+<account_id>    : “<home_account_id>-<environment>”
+<credential_id> : “<credential_type>-<client_id>-<realm>”
+<access_group>  : e.g. "com.microsoft.officecache"
+
+
+Type 1 (Secret shareable artifacts) Keychain Item Attributes
+============================================================
+ATTRIBUTE         VALUE
+~~~~~~~~~         ~~~~~~~~~~~~~~~~~~~~~~~~
+kSecClass         kSecClassGenericPassword
+kSecAttrAccount   <access_group>
+kSecAttrService   “Microsoft Credentials”
+kSecValueData     JSON data (UTF8 encoded) – shared credentials objects (multiple credentials saved in one keychain
+item)
+
+Type 2 (Non-secret shareable artifacts) Keychain Item Attributes
+================================================================
+ATTRIBUTE         VALUE
+~~~~~~~~~         ~~~~~~~~~~~~~~~~~~~~~~~~
+kSecClass         kSecClassGenericPassword
+kSecAttrAccount   <home_account_id>-<environment>-<realm>
+kSecValueData     JSON data (UTF8 encoded) – account object
+
+Type 3 (Secret non-shareable artifacts) Keychain Item Attributes
+===============================================================
+ATTRIBUTE         VALUE
+~~~~~~~~~         ~~~~~~~~~~~~~~~~~~~~~~~~
+kSecClass         kSecClassGenericPassword
+kSecAttrAccount   <access_group>-<app_bundle_id>-<home_account_id>-<environment>
+kSecAttrService   <credential_id>-<target>
+kSetAttrGeneric   <credential_id>
+kSecAttrType      Numeric Value: 2001=Access Token 2002=Refresh Token (Phase 1) 2003=IdToken
+kSecValueData     JSON data (UTF8 encoded) – credential object
+
+
+Type 1 JSON Data Example:
+{
+  "cache": {
+    "credentials": {
+      "account_id1": {
+        "credential_id1": "credential1 payload",
+        "credential_id2": "credential2 payload"
+      },
+      "account_id2": {
+        "credential_id1": "credential1 payload",
+        "credential_id2": "credential2 payload"
+      }
+    }
+  }
+}
+
+Additional Notes:
+* For a given <access_group>, multiple credentials are stored in
+  a single Type 1 keychain item.  This is a work-around for a macOS
+  keychain limitation related to ACLs (Access Control Lists) and
+  is intended to minimize macOS keychain access prompts.  Once
+  an application has access to the keychain item it can generally
+  access and update credentials without further keychain prompts.
+
+* Reference(s):
+  - Apple Keychain Services: https://developer.apple.com/documentation/security/keychain_services?language=objc
+  - Schema:
+https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview?path=%2FUnifiedSchema%2FSchema.md&version=GBdev
+
+*/
+
+@interface MSIDMacKeychainTokenCache ()
+@end
+
+@implementation MSIDMacKeychainTokenCache
+
+#pragma mark - init
+
+- (id)init {
+    self = [super init];
+    return self;
+}
+
+#pragma mark - Accounts
+
+// Write an account to the macOS keychain cache.
+- (BOOL)saveAccount:(MSIDAccountCacheItem *)account
```

Comment:
Since we just agreed that we will use the OID as the account id, can you add a check that the oid is present?